docs/cookbook.md

```diff
@@ -0,0 +1,363 @@
+custom_entities:
+    supplier:
+        options:
+            edit_form_extension: acme-supplier-edit-form
+```
+
+At this point, you should be able to dislay the edition form. But you should also see that all the fields are empty except the code. It’s because the entity has no normalizer, so we need to write one:
+
+```php
+<?php
+
+namespace Acme\Bundle\SupplierBundle\Normalizer;
+
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\scalar;
+use Acme\Bundle\SupplierBundle\Entity\Supplier;
+
+class SupplierNormalizer implements NormalizerInterface
+{
+    public function normalize($object, $format = null, array $context = array())
+    {
+        return [
+            'id' => $object->getId(),
+            'code' => $object->getCode(),
+            'name' => $object->getName(),
+        ];
+    }
+
+    public function supportsNormalization($data, $format = null)
+    {
+        return $data instanceof Supplier;
+    }
+
+}
+```
+
+The `normalize` method must return an array containing the reference data properties, with the property names as array keys and the property values as array values.
+
+Now declare the normalizer as a service:
+
+```yaml
+#Acme/Bundle/SupplierBundle/Resources/config/services.yml
+services:
+    acme.normalizer.supplier:
+        class: Acme\Bundle\SupplierBundle\Normalizer\SupplierNormalizer
+        tags:
+            - {name: pim_serializer.normalizer, priority: 200}
+```
+
+The service must be tagged as “pim_serializer.normalizer” to be identified as a serializer. You now have to load the 
+services.yml file via the dependency injection.
+
+At this point, you should be able to see all the fields of your reference data. If you click on the save button, the reference data should be saved and the list should also display the updated informations.
+
+#### Add the deletion action
+
+Same as the edition form, we need to declare the delete route and button
+
+```yaml
+#Acme/Bundle/SupplierBundle/Resources/config/datagrid/supplier.yml
+datagrid:
+    supplier:
+        #[...]
+        actions:
+            delete:
+                type: delete
+                label: acme.supplier.this.delete
+                icon: trash
+                link: delete_link
+```
+```yaml
+#Acme/Bundle/SupplierBundle/Resources/config/datagrid/supplier.yml
+datagrid:
+    supplier:
+        #[...]
+        properties:
+            delete_link:
+                type: url
+                route: pim_customentity_rest_delete
+                params:
+                    - id
+                    - customEntityName
+```
+You should now be able to delete any reference data by clicking the delete icon on the list.
+
+At this point, the datagrid configuration is fully working, and you should be able to list, edit or delete any reference data value.
+
+## Internationalization
+
+You have to create 2 translation files. Here is an example for english:
```

Comment:
Why two? You should explain one is for standard Symfony translations, and the other for our JS front stack.

Reply:
The thing is I don't really know why... Especially since the 2 files are identical ! (at least in training the files we created were strictly identical). Could you explain it to me then ?